### PR TITLE
Add hls-1.6.0.0

### PR DESCRIPTION
**Pull request by jneira**

### CODE REVIEW

**Comment by jneira** [comment]
@hasufell ci is happy 😄

**Comment by hasufell** [comment]
```
[ Info  ] verifying digest of: haskell-language-server-Linux-alpine-x86_64-1.6.0.0.tar.xz
ghcup-gen: Codec.Compression.Lzma.decompressWith: decoding error LzmaRetFormatError
```

**Comment by jneira** [comment]
is that a local check? i cant find that in the ci logs, it seems ci only checks own ghcup tarballs

**Comment by hasufell** [comment]
See https://github.com/haskell/ghcup-metadata#adding-a-new-ghc-version step 7

**Comment by jneira** [comment]
thanks! i inferred it from the ci invocation and it is running now

does it take too much time to check it in ci?

**Comment by hasufell** [comment]
> does it take too much time to check it in ci?

How would you run it?

**Comment by jneira** [comment]
hmm adding a step running `cabal run ghcup-gen -- check-tarballs -f ghcup-0.0.6.yaml -u 'hls-1\.6\.0\.0'`?

**Comment by hasufell** [comment]
> hmm adding a step running `cabal run ghcup-gen -- check-tarballs -f ghcup-0.0.6.yaml -u 'hls-1\.6\.0\.0'`?

How  will that work for the next PR adding a new HLS version or a different GHC version?

**Comment by jneira** [comment]
ok reproduced the check error locally but i still am not sure what is going on, i can extract the files succesfully. The unique thing i did locally was rename the file, so ci produced the tarball

**Comment by hasufell** [comment]
> ok reproduced the check error locally but i still am not sure what is going on, i can extract the files succesfully. The unique thing i did locally was rename the file, so ci produced the tarball

We use lzma to decompress:

https://gitlab.haskell.org/haskell/ghcup-hs/-/blob/655ee432f8330d426199ad6ece190fc194f860eb/lib/GHCup/Utils.hs#L716-719

https://hackage.haskell.org/package/lzma-static

It appears to have problem with some feature of the archive, I guess.

**Comment by jneira** [comment]
The encoding for `haskell-language-server-macOS-aarch64-1.6.0.0.tar.xz` for example is `LZMA2:23 CRC64`, for alpine is `LZMA:23`
So each tarball is generated in the system where hls is being built and the version of the tool can affect the final result 😟

**Comment by jneira** [comment]
I am thinking in remove the alpine tarballs, like 1.5.1. At the end those binaries will suffer the same template haskell bug we hit in hls , no?

**Comment by jneira** [comment]
In fact all tarballs are `LZMA2:23 CRC64` but the alpine one

File: ghcup-0.0.6.yaml [hunk 1]
```diff
@@ -2730,6 +2728,53 @@ ghcupDownloads:
...
+              dlHash: 736c5d286d0a2d488baa2ef68961cdad8846410acb423115c2f42a0a0cf46dcd
+            '( >= 13 )':
+              dlUri: https://github.com/haskell/haskell-language-server/releases/download/1.6.0.0/haskell-language-server-FreeBSD13-x86_64-1.6.0.0.tar.xz
+              dlHash: 0fcb58a457ee91cd89011a3a49d6cedea9f2a002fcdd4b0bbd4fe1d550b9bdf7
+          Windows:
+            unknown_versioning:
+              dlUri: https://github.com/haskell/haskell-language-server/releases/download/1.6.0.0/haskell-language-server-Windows-1.6.0.0.tar.gz
+              dlHash: 9f1b4ea31602a775a8f5bfedcc6290d3e45dd695f389fd705c9fff96c63c8570
+          Linux_Alpine:
+            unknown_versioning: *hls-1600-64
```
Comment (hasufell) [inline ghcup-0.0.6.yaml:2763]: If the current linux binaries are dynamic, this will not work.

**Pull request merged**